x-pack/plugins/ml/common/util/job_utils.ts

```diff
@@ -85,6 +73,33 @@ export function isMappableJob(job: CombinedJob, detectorIndex: number): boolean
   return isMappable;
 }
 
+/**
+ * Validates that composite definition only have sources that are only terms and date_histogram
+ * if composite is defined.
+ * @param buckets
+ */
+export function hasValidComposite(buckets: estypes.AggregationContainer) {
+  if (
+    isPopulatedObject(buckets, ['composite']) &&
+    isPopulatedObject(buckets.composite, ['sources']) &&
+    buckets.composite.sources !== undefined
```

Comment:
shall we check for an array instead? 
```suggestion
    Array.isArray(buckets.composite.sources)
```

Reply:
Updated here [`4987fc2` (#96348)](https://github.com/elastic/kibana/pull/96348/commits/4987fc22ff2cf6f338f3bbeffa330dcf613290ff)